File: client-examples/go/smartcompress.go
package smartcompress

import (
	"bytes"
	"fmt"
	"io"
	"mime/multipart"
	"net/http"
	"os"
	"strconv"
)

type Client struct {
	ApiKey string
	ApiUrl string
}

type Options struct {
	Quality int
	Width   int
	Height  int
	Format  string
}

func NewClient(apiKey string) *Client {
	return &Client{
		ApiKey: apiKey,
		ApiUrl: "https://api.shrinkix.com/compress",
	}
}

func (c *Client) FromFile(inputPath string, outputPath string, opts *Options) error {
	file, err := os.Open(inputPath)
	if err != nil {
		return err
	}
	defer file.Close()

	body := &bytes.Buffer{}
	writer := multipart.NewWriter(body)
	part, err := writer.CreateFormFile("image", inputPath)
	if err != nil {
		return err
	}
	io.Copy(part, file)

	if opts != nil {
		if opts.Quality > 0 {
			writer.WriteField("quality", strconv.Itoa(opts.Quality))
		}
		if opts.Width > 0 {
			writer.WriteField("width", strconv.Itoa(opts.Width))
		}
		if opts.Height > 0 {
			writer.WriteField("height", strconv.Itoa(opts.Height))
		}
		if opts.Format != "" {
			writer.WriteField("format", opts.Format)
		}
	}

	writer.Close()

	req, err := http.NewRequest("POST", c.ApiUrl, body)
	if err != nil {
		return err
	}
	req.Header.Set("X-API-Key", c.ApiKey)
	req.Header.Set("Content-Type", writer.FormDataContentType())

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		return fmt.Errorf("Compression failed with status: %s", resp.Status)
	}

	out, err := os.Create(outputPath)
	if err != nil {
		return err
	}
	defer out.Close()
	io.Copy(out, resp.Body)

	return nil
}

File: client-examples/python/smartcompress.py
import requests
import os

class SmartCompress:
    def __init__(self, api_key):
        self.api_key = api_key
        self.api_url = "https://api.shrinkix.com/compress"

    def from_file(self, input_path, output_path, options=None):
        if options is None:
            options = {}

        if not os.path.exists(input_path):
            raise FileNotFoundError(f"File not found: {input_path}")

        files = {
            'image': open(input_path, 'rb')
        }
        
        headers = {
            'X-API-Key': self.api_key
        }
        
        # Prepare data for other fields
        data = {}
        if 'quality' in options: data['quality'] = options['quality']
        if 'width' in options: data['width'] = options['width']
        if 'height' in options: data['height'] = options['height']
        if 'format' in options: data['format'] = options['format']

        try:
            response = requests.post(self.api_url, files=files, headers=headers, data=data)
            
            # Close file
            files['image'].close()

            if response.status_code == 200:
                with open(output_path, 'wb') as f:
                    f.write(response.content)
                return True
            else:
                raise Exception(f"Compression failed: {response.text}")

        except Exception as e:
            if 'image' in files and not files['image'].closed:
                files['image'].close()
            raise e
